mne/io/eeglab/eeglab.py

```diff
@@ -270,21 +336,42 @@ def __init__(self, input_fname, montage, preload=False, eog=(),
         else:
             if preload is False or isinstance(preload, string_types):
                 warn('Data will be preloaded. preload=False or a string '
-                     'preload is not supported when the data is stored in the '
-                     '.set file')
+                     'preload is not supported when the data is stored in '
+                     'the .set file')
             # can't be done in standard way with preload=True because of
             # different reading path (.set file)
-            data = eeg.data.reshape(eeg.nbchan, -1, order='F')
-            data = data.astype(np.double)
+            n_chan, n_times = eeg.data.shape
+            data = np.empty((n_chan + 1, n_times), dtype=np.double)
+            data[:-1] = eeg.data
             data *= CAL
```

Comment:
I just gave it a try. Works without any errors. I mentioned a few issues. Good to go from my end once you address them